src/runtime_src/core/tools/xbmgmt2/OO_Config.cpp

```diff
@@ -19,32 +19,259 @@
+{
+  dev=dev;
+  std::cout << "Show device config\n";
+}
+
+/*
+ * helper function for option:daemon
+ * change host name in config
+ */
+static void 
+update_daemon_config(const std::string host)
+{
+  std::cout << "update config with " << host;
+}
+
+/*
+ * helper function for option:device
+ * update device configuration
+ */
+static void 
+update_device_conf(std::shared_ptr<xrt_core::device>& dev, 
+  const std::string lvl, configType config_type)
+{
+  dev=dev;
+  config_type=config_type;
+  std::cout << "update device config with level " << lvl;
+}
+
+/*
+ * helper function for option:en(dis)able_retention
+ * set data retention
+ */
+static void 
+memory_retention(std::shared_ptr<xrt_core::device>& dev, 
+  memType type, bool enable)
+{
+  dev=dev;
+  type=type;
+  enable=enable;
+  std::cout << "update memorty retention\n";
+}
+
+}
+//end anonymous namespace
+
+
 // ----- C L A S S   M E T H O D S -------------------------------------------
 
 OO_Config::OO_Config( const std::string &_longName)
     : OptionOptions(_longName, "<Add description>")
     , m_device("")
     , m_help(false)
+    , daemon(false)
+    , host("")
+    , security("")
+    , clk_scale("")
+    , power_override("")
+    , show(false)
+    , ddr(false)
+    , hbm(false)
+    , enable_retention(false)
+    , disable_retention(false)
+
 {
   m_optionsDescription.add_options()
     ("device,d", boost::program_options::value<decltype(m_device)>(&m_device), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
-    ("daemon", "<add description>")
-    ("host", "<add description>")
-    ("security", "<add description>")
-    ("show", "<add description>")
+    ("daemon", boost::program_options::bool_switch(&daemon), "<add description>")
+    ("host", boost::program_options::value<decltype(host)>(&host), "ip or hostname for peer")
+    ("security", boost::program_options::value<decltype(security)>(&security), "<add description>")
+    ("runtime_clk_scale", boost::program_options::value<decltype(clk_scale)>(&clk_scale), "<add description>")
+    ("cs_threshold_power_override", boost::program_options::value<decltype(power_override)>(&power_override), "<add description>")
+    ("show", boost::program_options::bool_switch(&show), "<add description>")
+    ("enable_retention", boost::program_options::bool_switch(&enable_retention), "<add description>")
+    ("disable_retention", boost::program_options::bool_switch(&disable_retention), "<add description>")
+    ("ddr", boost::program_options::bool_switch(&ddr), "<add description>")
+    ("hbm", boost::program_options::bool_switch(&hbm), "<add description>")
     ("help,h", boost::program_options::bool_switch(&m_help), "Help to use this sub-command")
   ;
 }
 
 void
 OO_Config::execute(const SubCmdOptions& _options) const
 {
-  _options.size();
-  printHelp();
+
+  XBU::verbose("SubCommand option: config");
+
+  XBU::verbose("Option(s):");
+  for (auto & aString : _options) {
+    std::string msg = "   ";
```

Comment:
Since this code is simply add some spaces in front of the string, can you refactor it to:
1) Either use a boost::format(), which might be too much.
OR use
2) XBU::verbose(std::string("   ") + aString);